目录加密-目录输入密码访问-看不见文章

### 是什么版本出现了此问题？

1.4.11

### 使用的什么数据库？

H2

### 使用的哪种方式部署？

Docker

### 在线站点地址

https://www.ycdr.fun/

### 发生了什么？

1. 设置加密目录-目录访问文章 列表为空
2. 问题分析 逻辑有问题
```java
//run.halo.app.controller.content.model.CategoryModel#listPost  85

//该方法请求目录列表 status 状态是写死的 为 0  PUBLISHED
// 应该增加判断 如果目录有密码 查询条件 应该是 0 3 
//我本地修改后没有此问题了

        final Category category = categoryService.getBySlugOfNonNull(slug, true);
        if (!authenticationService.categoryAuthentication(category.getId(), null)) {
            model.addAttribute("slug", category.getSlug());
            model.addAttribute("type", EncryptTypeEnum.CATEGORY.getName());
            if (themeService.templateExists(POST_PASSWORD_TEMPLATE + SUFFIX_FTL)) {
                return themeService.render(POST_PASSWORD_TEMPLATE);
            }
            return "common/template/" + POST_PASSWORD_TEMPLATE;
        }
       // 此处为我的修改
        Set<PostStatus> status=Sets.immutableEnumSet(PostStatus.PUBLISHED);
        if (StringUtils.isNotBlank(category.getPassword())) {
            status = Sets.immutableEnumSet(PostStatus.PUBLISHED,PostStatus.INTIMATE);
        }
      //修改结束
        CategoryDTO categoryDTO = categoryService.convertTo(category);

        final Pageable pageable = PageRequest.of(page - 1,
            optionService.getArchivesPageSize(),
            Sort.by(DESC, "topPriority", "createTime"));
        Page<Post> postPage =
            postCategoryService.pagePostBy(category.getId(), status, pageable);
        Page<PostListVO> posts = postService.convertToListVo(postPage);

```

3. 不清楚是否是我使用错误 
4. 主题使用的是雪
5. 我的网站 每日计划 为加密目录 密码为1 ，文章有四篇 目前显示不出来
### 相关日志输出

_No response_

### 附加信息

_No response_

com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module LineLength - Property 'fileExtensions' does not exist, please check the documentation
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:477)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:61)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:26)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:60)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:51)
	at org.infernus.idea.checkstyle.checker.CheckerFactoryWorker.run(CheckerFactoryWorker.java:46)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property 'fileExtensions' does not exist, please check the documentation
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:223)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:191)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:472)
	... 7 more

com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module LineLength - Property 'fileExtensions' does not exist, please check the documentation
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:477)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:61)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:26)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:60)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:51)
	at org.infernus.idea.checkstyle.checker.CheckerFactoryWorker.run(CheckerFactoryWorker.java:46)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property 'fileExtensions' does not exist, please check the documentation
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:223)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:191)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:472)
	... 7 more

--- a/src/main/java/run/halo/app/controller/content/model/CategoryModel.java
+++ b/src/main/java/run/halo/app/controller/content/model/CategoryModel.java
@@ -96,9 +96,9 @@ public class CategoryModel {
             return "common/template/" + POST_PASSWORD_TEMPLATE;
         }
 
-        Set<PostStatus> status = Sets.immutableEnumSet(PostStatus.PUBLISHED);
+        Set<PostStatus> statuses = Sets.immutableEnumSet(PostStatus.PUBLISHED);
         if (StringUtils.isNotBlank(category.getPassword())) {
-            status = Sets.immutableEnumSet(PostStatus.INTIMATE);
+            statuses = Sets.immutableEnumSet(PostStatus.INTIMATE);
         }
 
         CategoryDTO categoryDTO = categoryService.convertTo(category);
@@ -107,7 +107,7 @@ public class CategoryModel {
             optionService.getArchivesPageSize(),
             Sort.by(DESC, "topPriority", "createTime"));
         Page<Post> postPage =
-            postCategoryService.pagePostBy(category.getId(), status, pageable);
+            postCategoryService.pagePostBy(category.getId(), statuses, pageable);
         Page<PostListVO> posts = postService.convertToListVo(postPage);
 
         // Generate meta description.

--- a/src/main/java/run/halo/app/controller/content/model/CategoryModel.java
+++ b/src/main/java/run/halo/app/controller/content/model/CategoryModel.java
@@ -5,6 +5,7 @@ import static run.halo.app.model.support.HaloConst.POST_PASSWORD_TEMPLATE;
 import static run.halo.app.model.support.HaloConst.SUFFIX_FTL;
 
 import com.google.common.collect.Sets;
+import java.util.Set;
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageRequest;
@@ -77,8 +78,8 @@ public class CategoryModel {
      * List category posts.
      *
      * @param model model
-     * @param slug slug
-     * @param page current page
+     * @param slug  slug
+     * @param page  current page
      * @return template name
      */
     public String listPost(Model model, String slug, Integer page) {
@@ -95,14 +96,18 @@ public class CategoryModel {
             return "common/template/" + POST_PASSWORD_TEMPLATE;
         }
 
+        Set<PostStatus> status = Sets.immutableEnumSet(PostStatus.PUBLISHED);
+        if (StringUtils.isNotBlank(category.getPassword())) {
+            status = Sets.immutableEnumSet(PostStatus.INTIMATE);
+        }
+
         CategoryDTO categoryDTO = categoryService.convertTo(category);
 
         final Pageable pageable = PageRequest.of(page - 1,
             optionService.getArchivesPageSize(),
             Sort.by(DESC, "topPriority", "createTime"));
         Page<Post> postPage =
-            postCategoryService.pagePostBy(category.getId(), Sets
-                .immutableEnumSet(PostStatus.PUBLISHED), pageable);
+            postCategoryService.pagePostBy(category.getId(), status, pageable);
         Page<PostListVO> posts = postService.convertToListVo(postPage);
 
         // Generate meta description.

